PiwikTracker/URLSessionDispatcher.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+final class URLSessionDispatcher: Dispatcher {
+    
+    let timeout: TimeInterval
+    let session: URLSession
+    let baseURL: URL
+    
+    var userAgent: String? = {
+        let webView = UIWebView(frame: .zero)
+        let currentUserAgent = webView.stringByEvaluatingJavaScript(from: "navigator.userAgent") ?? ""
+        return currentUserAgent.appending(" Piwik iOS SDK URLSessionDispatcher")
+    }()
+    
+    init(baseURL: URL) {
+        if !baseURL.absoluteString.hasSuffix("piwik.php") {
+            fatalError("The baseURL is expected to end in piwik.php")
+        }
+        self.baseURL = baseURL
+        self.timeout = 5
+        self.session = URLSession.shared
+    }
+    
+    func send(event: Event, success: @escaping ()->(), failure: @escaping (_ shouldContinue: Bool)->()) {
+        let url = baseURL.setting(event.queryItems)
+        var request = URLRequest(url: url, cachePolicy: .reloadIgnoringCacheData, timeoutInterval: timeout)
+        request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
+        request.httpMethod = "GET"
+        send(request: request, success: success, failure: failure)
+    }
+    
+    func send(events: [Event], success: @escaping ()->(), failure: @escaping (_ shouldContinue: Bool)->()) {
+        let eventsAsQueryItems = events.map({ event in event.queryItems })
+        let serializedEvents = eventsAsQueryItems.map({ items in
+            items.flatMap({ item in
+                guard let value = item.value,
+                    let encodedValue = value.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else { return nil }
+                return "\(item.name)=\(encodedValue)"
+            }).joined(separator: "&")
+        })
+        let body = ["requests": serializedEvents.map({ "?\($0)" })]
+        guard let jsonBody = try? JSONSerialization.data(withJSONObject: body, options: []) else {
+            puts("Unable to serialize JSONData")
+            failure(false)
+            return
+        }
+        var request = URLRequest(url: baseURL, cachePolicy: .reloadIgnoringCacheData, timeoutInterval: timeout)
+        request.setValue("application/json; charset=utf-8", forHTTPHeaderField: "Content-Type")
+        request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
+        request.httpMethod = "POST"
+        request.httpBody = jsonBody
+        send(request: request, success: success, failure: failure)
+    }
+    
+    private func send(request: URLRequest, success: @escaping ()->(), failure: @escaping (_ shouldContinue: Bool)->()) {
+        let task = session.dataTask(with: request) { data, response, error in
+            // should we check the response?
+            // let dataString = String(data: data!, encoding: String.Encoding.utf8)
+            if error == nil {
```

Comment:
As far as I know, the result is only interesting when debugging the sdk. The information it returns possibly helps finding sdk-issues.

What do you think about having a debug property in the Tracker itself which is false per default? If it is set, we parse the data and will print it to the console.